File: exercise/scala/day11/src/main/scala/christmas/Preparation.scala
package christmas

import christmas.Preparation.ToyType.*

object Preparation {
  def prepareGifts(numberOfGifts: Int): String = {
    if (numberOfGifts <= 0) "No gifts to prepare."
    else if (numberOfGifts < 50) "Elves will prepare the gifts."
    else "Santa will prepare the gifts."
  }

  def categorizeGift(age: Int): String = {
    if (age <= 2) "Baby"
    else if (age <= 5) "Toddler"
    else if (age <= 12) "Child"
    else "Teen"
  }

  def ensureToyBalance(toyType: ToyType, toysCount: Int, totalToys: Int): Boolean = {
    val typePercentage = toysCount.toDouble / totalToys
    toyType match {
      case EDUCATIONAL => typePercentage >= 0.25
      case FUN => typePercentage >= 0.30
      case CREATIVE => typePercentage >= 0.20
    }
  }

  enum ToyType {
    case EDUCATIONAL, FUN, CREATIVE
  }
}

File: exercise/php/day11/src/Christmas/Preparation.php
<?php

namespace Christmas;

class Preparation {
    public static function prepareGifts(int $numberOfGifts): string {
        return match (true) {
            $numberOfGifts <= 0 => 'No gifts to prepare.',
            $numberOfGifts < 50 => 'Elves will prepare the gifts.',
            default => 'Santa will prepare the gifts.'
        };
    }

    public static function categorizeGift(int $age): string {
        return match (true) {
            $age <= 2 => 'Baby',
            $age <= 5 => 'Toddler',
            $age <= 12 => 'Child',
            default => 'Teen'
        };
    }

    public static function ensureToyBalance(ToyType $toyType, int $toysCount, int $totalToys): bool {
        $typePercentage = $toysCount / $totalToys;

        return match ($toyType) {
            ToyType::EDUCATIONAL => $typePercentage >= 0.25,
            ToyType::FUN => $typePercentage >= 0.30,
            ToyType::CREATIVE => $typePercentage >= 0.20,
        };
    }
}
